### PR TITLE
Refactor Viewer code and addition of volume selector

### PR DESCRIPTION
**Pull request by brisvag**
Volume selector is here!

Now `p.show()` will also automatically dock a simple dropdown menu widget in the napari viewer. This widget lets you select which volume to show!

There's also a couple of minor bufixes thrown in.

### CODE REVIEW

File: peepingtom/gui/widgets/volume_selector.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        if self.volumes:
+            self.select_from_dropdown()
+
+    def _init_gui(self):
+        self.widget = QWidget()
+        layout = QVBoxLayout()
+        self.widget.setLayout(layout)
+
+        layout.addWidget(QLabel('PeepingTom - Volume Selector'))
```
Comment (alisterburt) [inline peepingtom/gui/widgets/volume_selector.py:19]: 'PeepingTom - Volume Selector' is used here and in the `add_dock_widget` call - refactor into `VolumeSelector.name`?

**Review by alisterburt — APPROVED** [review]
Looks great! Really excited about this - one tiny point to address then let's get this in!!

File: peepingtom/gui/widgets/volume_selector.py (hunk 1 above)
Comment (brisvag) [reply peepingtom/gui/widgets/volume_selector.py]: Ok, did it, more or less!

**Comment by alisterburt** [comment]
non-obvious fail in tests, does it happen when you test locally?

**Comment by brisvag** [comment]
@alisterburt I'm trying to fix it... it's a qt problem, almost certainly. Locally it's fine, but running graphical programs on the github actions VMs seems to be a bit tricky!

**Comment by alisterburt** [comment]
uh oh! Have you looked at the qtbot fixture?
https://pytest-qt.readthedocs.io/en/latest/intro.html
That's what's used in napari for qt testing

**Comment by brisvag** [comment]
I decided to drop testing of qt stuff, for now. It seems to be the same as [this issue](https://github.com/The-Compiler/pytest-xvfb/issues/11), which has no solution yet.
The problem actually happened locally as well, I was just doing things wrong at first. The thing is, tests pass just fine, but then the command fails with an exit code of `1` because some qt stuff breaks after all the tests are already finished.

To keep codecov happy, I added some unrelated tests for now, so we can merge this :P

**Comment by brisvag** [comment]
Qt tests are up and running, thanks to the `make_napari_viewer` pytest fixture!

**Pull request merged**